utils/land-patterns.stanza

```diff
@@ -1040,82 +1040,165 @@ public pcb-landpattern smd-testpoint-pkg (d:Double) :
+  IpcResults(Zmax, Gmin, Xmin)
+   
+; Create an `n` pin SOIC land pattern. 
+;
+; Restrictions:
+; - n must be even
+; - the leads are oriented on the left and right sides of the land pattern
+; - p[1] is the top left pin
+;
+public defn make-n-pin-soic-landpattern (num-pins:Int,       ; number of pins of the component
+                                         pitch:Double,       ; pitch of the pins of the component
+                                         outer-x:Toleranced, ; the "outer length" (x-dimension) of the component, from lead-to-lead
+                                         part-x:Toleranced,  ; the length of the part/package (x-dimension)
+                                         part-y:Toleranced,   ; the width of the part/package (y-dimension)
+                                         lead-type:LandProtrusionType, ; the lead/land protrusion type of the part
+                                         lead-x:Toleranced, ; the length (x-dimension) of the leads
+                                         lead-y:Toleranced,  ; the width (y-dimension) of the leads
+                                         density-level:DensityLevel):
+  fatal("n must be even") when (num-pins % 2) != 0
+  val lead-fillets = lead-fillets(lead-type, density-level)
+
+  inside pcb-landpattern:
+    ;----------------------------------------------------
+    ; Compute adjustments to dimensions using IPC formula
+    val ipc = ipc-formula(
+      outer-x, lead-x, lead-y, 
+      lead-fillets
+    )
+
+    ;----------------------------------------------------
+    ; compute our pad sizes
+    val pad-sz  = pad-size(ipc)
+
+    ;----------------------------------------------------
+    ; place the pads
+    val grid-sz = Dims(Gmin(ipc) + x(pad-sz), pitch)
+    val locs    = to-tuple(grid-locs(num-pins / 2, 2, x(grid-sz), y(grid-sz), false))
+    for (i in 0 to false, l in locs) do:
+      val r = i % (num-pins / 2)
+      val c = i / (num-pins / 2)
+      val idx = (i + 1) when c == 0 else num-pins - r
+      pad p[idx] : smd-pad(Rectangle(pad-sz)) at l
+    
+    ;----------------------------------------------------
+    ; compute the courtyard
+    val [max-y, min-y] = fork-on-seq(seq(y{center(_)}, locs), maximum, minimum)
+    val lp-y-dim = y(pad-sz) + max-y - min-y
+    val lp-sz   = enlarge(Dims(Zmax(ipc), y(pad-sz) + max-y - min-y), 2.0 * SOLDER-MASK-REGISTRATION)
+    val part-sz = Dims(max-value(part-x), max-value(part-y))
+    val cy-sz   = enlarge(max(lp-sz, part-sz), 2.0 * courtyard-excess(lead-fillets))
+    layer(Courtyard(Top)) = Rectangle(cy-sz)
+        
+    ;----------------------------------------------------
+    ; draw the orientation marker
+    val line-width  = clearance(current-rules(), MinSilkscreenWidth)
+    val line-length = 3.0 * line-width - line-width
+    val dist  = SOLDER-MASK-REGISTRATION + SILK-SOLDER-MASK-CLEARANCE + line-width * 0.5 + 0.01
```

Comment:
minor, but you can move this just before `layer(Silkscreen(...`

Reply:
done